server/rootless.go

```diff
@@ -17,7 +21,41 @@ func hasNetworkNamespace(config *rspec.Spec) bool {
 }
 
 func makeOCIConfigurationRootless(g *generate.Generator) {
-	g.Config.Linux.Resources = nil
+	// Resource limitations requires cgroup v2 delegation (https://rootlesscontaine.rs/getting-started/common/cgroup2/).
+	if g.Config.Linux.Resources != nil {
+		// cannot control device eBPF with rootless
+		g.Config.Linux.Resources.Devices = nil
```

Comment:
Do we need a warning here (like you do in the other cases below)?

Reply:
No, because there is no action user can take.